File: me_daily/2025/10/1020/cf2148.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"slices"
	"strconv"
	"strings"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func readLine() string {
	s, _ := reader.ReadString('\n')
	return strings.TrimSpace(s)
}

var (
	t, n int
	a    []int64
)

func readInts() []int {
	fields := strings.Fields(readLine())
	res := make([]int, len(fields))
	for i, f := range fields {
		res[i], _ = strconv.Atoi(f)
	}
	return res
}
func solve(a []int64) int64 {
	var sumEven int64 = 0
	oddVals := []int64{}
	for _, x := range a {
		if x%2 == 0 {
			sumEven += x
		} else {
			oddVals = append(oddVals, x)
		}
	}
	k := len(oddVals)
	if k == 0 {
		return 0
	}
	var sumOdd int64
	slices.Sort(oddVals)
	for _, x := range oddVals {
		sumOdd += x
	}
	total := sumEven + sumOdd
	numSac := k / 2 // Go的int除法对正数就是向下取整 (floor)
	var sumSac int64
	for i := 0; i < numSac; i++ {
		sumSac += oddVals[i]
	}
	return total - sumSac
}
func main() {
	defer writer.Flush() //main() 结束时自动调用
	t, _ := strconv.Atoi(readLine())
	for ; t > 0; t-- {
		n, _ := strconv.Atoi(readLine())
		a = make([]int64, n)
		readInt := readInts()
		for i := 0; i < n; i++ {
			a[i] = int64(readInt[i])
		}
		fmt.Fprintln(writer, solve(a))
	}

}

File: me_daily/2025/10/1020/cf2148.rs
use std::io::{self, Read};

fn solve(a: &Vec<i64>) -> i64 {
    let mut sum_even = 0i64;
    let mut odd_vals: Vec<i64> = Vec::new();
    for &x in a {
        if x % 2 == 0 {
            sum_even += x;
        } else {
            odd_vals.push(x);
        }
    }
    let k = odd_vals.len();
    if k == 0 {
        return 0;
    }
    odd_vals.sort_unstable(); // 升序，小的在前好牺牲
    let sum_odd: i64 = odd_vals.iter().sum();
    let num_sac = k / 2; // floor(k/2)
    let sum_sac: i64 = odd_vals.iter().take(num_sac).sum();
    sum_even + sum_odd - sum_sac
}
fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut iter = input.split_whitespace();

    let mut next = || iter.next().unwrap();

    let t: usize = next().parse().unwrap();
    for _ in 0..t {
        let n: usize = next().parse().unwrap();
        let a: Vec<i64> = (0..n).map(|_| next().parse().unwrap()).collect();
        println!("{}", solve(&a));
    }
}
